server/src/main/java/org/opensearch/index/codec/composite/composite99/Composite99DocValuesWriter.java

```diff
@@ -76,6 +90,76 @@ public Composite99DocValuesWriter(DocValuesConsumer delegate, SegmentWriteState
+                segmentWriteState.segmentInfo.name,
+                DocIdSetIterator.NO_MORE_DOCS,
+                segmentWriteState.segmentInfo.getUseCompoundFile(),
+                segmentWriteState.segmentInfo.getHasBlocks(),
+                segmentWriteState.segmentInfo.getCodec(),
+                segmentWriteState.segmentInfo.getDiagnostics(),
+                segmentWriteState.segmentInfo.getId(),
+                segmentWriteState.segmentInfo.getAttributes(),
+                segmentWriteState.segmentInfo.getIndexSort()
+            );
+
+            SegmentWriteState consumerWriteState = new SegmentWriteState(
+                segmentWriteState.infoStream,
+                segmentWriteState.directory,
+                segmentInfo,
+                segmentWriteState.fieldInfos,
+                segmentWriteState.segUpdates,
+                segmentWriteState.context
+            );
+
+            this.composite99DocValuesConsumer = LuceneDocValuesConsumerFactory.getDocValuesConsumerForCompositeCodec(
+                consumerWriteState,
+                Composite99DocValuesFormat.DATA_DOC_VALUES_CODEC,
+                Composite99DocValuesFormat.DATA_DOC_VALUES_EXTENSION,
+                Composite99DocValuesFormat.META_DOC_VALUES_CODEC,
+                Composite99DocValuesFormat.META_DOC_VALUES_EXTENSION
+            );
+
+            String dataFileName = IndexFileNames.segmentFileName(
+                this.state.segmentInfo.name,
+                this.state.segmentSuffix,
+                Composite99DocValuesFormat.DATA_EXTENSION
+            );
+            dataOut = this.state.directory.createOutput(dataFileName, this.state.context);
+            CodecUtil.writeIndexHeader(
+                dataOut,
+                Composite99DocValuesFormat.DATA_CODEC_NAME,
+                Composite99DocValuesFormat.VERSION_CURRENT,
+                this.state.segmentInfo.getId(),
+                this.state.segmentSuffix
+            );
+
+            String metaFileName = IndexFileNames.segmentFileName(
+                this.state.segmentInfo.name,
+                this.state.segmentSuffix,
+                Composite99DocValuesFormat.META_EXTENSION
+            );
+            metaOut = this.state.directory.createOutput(metaFileName, this.state.context);
+            CodecUtil.writeIndexHeader(
+                metaOut,
+                Composite99DocValuesFormat.META_CODEC_NAME,
+                Composite99DocValuesFormat.VERSION_CURRENT,
+                this.state.segmentInfo.getId(),
+                this.state.segmentSuffix
+            );
+
+            success = true;
+        } finally {
+            if (success == false) {
+                IOUtils.closeWhileHandlingException(this);
+            }
+        }
         // check if there are any composite fields which are part of the segment
         // TODO : add integ test where there are no composite fields in a segment, test both flush and merge cases
```

Comment:
Same address and remove this comment ^

Reply:
what do you mean by composite fields? Like the whole composite field is missing or any dimension / metric is missing?

Reply:
As in missing composite fields like we have in abstractbuildertests. but we need to add such cases as integ tests

Reply:
I think StarTreeDocValueFormatTests looks after this case. I can add another one in the follow up PR to be sure.